src/core/net/tcp6.cpp

```diff
@@ -163,25 +232,48 @@ Error Tcp::Endpoint::Deinitialize(void)
     SuccessOrExit(error = tcp.mEndpoints.Remove(*this));
     SetNext(nullptr);
 
+    SuccessOrExit(error = Abort());
+
 exit:
     return error;
 }
 
 uint8_t Tcp::Endpoint::TimerFlagToIndex(uint8_t aTimerFlag)
 {
-    OT_UNUSED_VARIABLE(aTimerFlag);
-    /*
-     * TODO: Convert from the timer flag provided by TCPlp to the index in
-     * our timers array.
-     */
-    return 0;
+    switch (aTimerFlag)
+    {
+    case TT_DELACK:
+        return kTimerDelack;
+    case TT_REXMT:
+        /* fallthrough intentional */
```

Comment:
Use `OT_FALL_THROUGH` instead (to avoid some toolchain warnings/error)?

Or actually in this case, it may be fine to use
```
  case TT_REXMT:
  case TT_PERSIST:
       ...
```